### PR TITLE
Add Referrals

### PR DESCRIPTION
**Pull request by jguitar**
- Added ability to create and list referrals for the pet parent
- Added referral support to fake api
- Added webmock and fake api specs

### CODE REVIEW

File: lib/kb/fake/bounded_context/pet_family/pet_parents.rb [hunk 1]
```diff
@@ -26,6 +26,18 @@ def petparents_filterable_attributes
...
         end
 
+        get '/v1/petparents/:key/referrals' do
+          json_response 200, referrals_by_pet_parent_key(params['key'])
+        end
+
+        post '/v1/petparents/:key/referrals' do
+          resource = JSON.parse(request.body.read)
+          resource = resource.merge 'key' => SecureRandom.uuid
+          resource = resource.merge 'petParentKey' => params['key']
```
Comment (fatbeard2) [inline lib/kb/fake/bounded_context/pet_family/pet_parents.rb:36]: Is this attribute going to be present in the "real" response from kb? Or is it added here only for the purposes of filtering in `referrals_by_pet_parent_key` method?

I'm thinking that if it's going to be present in the real response - we could add it to the list of `attributes` of `Referral`.

**Comment by jguitar** [comment]
I've updated gem with the last changes but still there are some pending changes from the KB.
Currently KB is not responding with the object after the referral creation.

File: barkibu-kb.gemspec [hunk 2]
```diff
@@ -51,4 +51,5 @@ Gem::Specification.new do |spec|
   spec.add_runtime_dependency 'faraday-http'
   spec.add_runtime_dependency 'faraday_middleware'
   spec.add_runtime_dependency 'i18n'
+  spec.add_runtime_dependency 'webmock'
```
[the remark this answers is not in the document]
Comment (fatbeard2) [reply barkibu-kb.gemspec:54]: Although I haven't used this 2 gem setup before. Not sure if it's even needed to be specified here as other dev dependencies are not. Should all dev deps be specified only in fake gem?

[the remark this answers is not in the document]
Comment (malkovro) [reply barkibu-kb.gemspec:54]: Needed per say no, but as we make use of it explicitly i think it makes sense to reference the dependency explicitly as well.

**Review by malkovro — APPROVED** [review]

**Pull request merged**